sphinx/util/inspect.py

```diff
@@ -130,7 +132,10 @@ def safe_repr(object):
     except Exception:
         raise ValueError
     if isinstance(s, binary_type):
-        return force_decode(s, None).replace('\n', ' ')
+        s = force_decode(s, None)
+    # Strip non-deterministic memory addresses such as
+    # ``<__main__.A at 0x7f68cb685710>``
+    s = re.sub(r' at 0x[0-9a-f]{8,12}(?=>$)', '', s)
```

Comment:
actually couldn't 64-bit addresses be up to 16 chars long? also the RE should be precompiled.